contribs/gmf/src/directives/layertree.js

```diff
@@ -224,10 +225,23 @@ gmf.LayertreeController.prototype.prepareLayer_ = function(node, layer) {
 
   // If layer is 'unchecked', set it to invisible.
   var metadata = node.metadata;
-  if (node.children === undefined && goog.isDefAndNotNull(metadata)) {
-    if (!metadata['isChecked']) {
+  if (isMerged) {
+    //Case Non Mixed group -> Hide the layer if all child nodes have isCheck set to false
+    this.getFlatNodes_(node, childNodes);
+    childNodesUnchecked = childNodes.filter(function(childNode) {
+      return !childNode.metadata || !childNode.metadata['isChecked'];
+    });
```

Comment:
You could also use https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/every which would directy return a boolean.